Iteration on exisiting arbitrate section

fleupold opened the pull request:
# Description

Another pass over the existing sections to make them sound a bit more a like. Main focus was on the first section as this one was still referencing the driver and solver in inconsistent ways

# Changes

- [x] Move driver/solver engine chapters into a solver subfolder (to make the structure reflect the fact that solver is one component consisting of two subcomponents).
- [x] Rewrite top level summary to reflect solver = driver + engine logic
- [x] Small iterations to make the whole text sound more like written from one voice

<!--
## Related Issues

Fixes #
-->

fleupold committed:
Iteration on exisiting arbitrate section

fleupold committed:
add category file for new section

MartinquaXD commented on docs/cow-protocol/tutorials/arbitrate/README.mdx:
```diff
@@ -1,35 +1,46 @@
+## Architecture
 
-#### The Orderbook
+Four main sub-systems are required to facilitate and drive the auction.
```
I would say we should consider `solvers` (i.e. `driver` + `engine`) a single entity because their split is just an implementation detail. Also your choice of headlines would suggest the same (driver and engine are a section under the main component `solvers`)
```suggestion
Three main sub-systems are required to facilitate and drive the auction.
```

MartinquaXD commented on docs/cow-protocol/tutorials/arbitrate/README.mdx:
```diff
@@ -1,35 +1,46 @@
-It prepares all the data needed for the solvers to solve the auction and sends it to them.
-It also receives solutions from solvers and does a postprocessing on them before sending them to the autopilot.
+This is the executive part of the solving process, responsible for interacting with the blockchain in order to enrich, encode and eventually execute the settlement.
+It prepares all the data needed for the engine to match order in the auction and forwards the augmented auction to the engine.
```
```suggestion
It prepares all the data needed for the engine to match orders in the auction and forwards the augmented auction to the engine.
```

MartinquaXD commented on docs/cow-protocol/tutorials/arbitrate/README.mdx:
```diff
@@ -1,35 +1,46 @@
+This is the executive part of the solving process, responsible for interacting with the blockchain in order to enrich, encode and eventually execute the settlement.
+It prepares all the data needed for the engine to match order in the auction and forwards the augmented auction to the engine.
+In turn it receives raw solutions from the solver engine, which it post-processes, simulates, merges, encodes into a blockchain transaction and finally scores.
 Postprocessing includes encoding, simulating, merging and ranking solutions.
```
All these tasks were already mentioned in the sentence before.
```suggestion
```

MartinquaXD reviewed (approved):
Just nits

fleupold committed:
Apply suggestions from code review

Co-authored-by: Martin Beckmann <martin.beckmann@protonmail.com>

mfw78 reviewed (approved):
Have an additional unifying commit in #244

mfw78 committed:
chore: polish / unify arbitrate (#244)

# Description

This PR applies further polish to the arbitration section for alignment.

# Changes

- [x] Ensure unified heading layout across section
- [x] Break some large paragraphs into lists instead
- [x] Standardise OrderBook capitalisation when used in standalone
- [x] Other minor editorial changes

---------

Co-authored-by: Martin Beckmann <martin.beckmann@protonmail.com>
